### PR TITLE
10 the add profile screen should open in a dialog instead of a new browser window

### PR DESCRIPTION
**Pull request by svdwyer**

### CODE REVIEW

File: src/popup/popup.html [hunk 1]
```diff
@@ -66,6 +66,52 @@ <h2 class="section-title">Your profiles</h2>
...
+
+            <fieldset>
+                <label for="idp">IDP</label>
+                <input id="idp" name="idp" type="text" placeholder="Solid identity provider" />
+                <p id="idp_error" class="error-explanation"></p>
+            </fieldset>
+
+            <fieldset>
+                <label for="webid">... or WebID</label>
+                <input id="webid" name="webid" type="text" placeholder="Your pod's WebID" />
```
[the remark this answers is not in the document]
Comment (svdwyer) [reply src/popup/popup.html:106]: So should I reword it to "Your WebID"?

File: src/popup/popup.js [hunk 2]
```diff
@@ -2,6 +2,93 @@ let availableIdentities = [];
...
+  idp: document.querySelector('#add-profile-dialog input[name=idp]'),
+  webID: document.querySelector('#add-profile-dialog input[name=webid]'),
+  colors: document.querySelectorAll('#add-profile-dialog input[name=color]'),
+  form: document.querySelector('#add-profile-dialog form'),
+  close: document.querySelector('#add-profile-dialog .close-button'),
+  avatar: document.querySelector('#add-profile-dialog .avatar'),
+  dialog: document.querySelector('#add-profile-dialog')
+};
+
+const initAddProfileScreen = () => {
```
[the remark this answers is not in the document]
Comment (svdwyer) [reply src/popup/popup.js:20]: ✔️

File: src/popup/popup.js [hunk 3]
```diff
@@ -2,6 +2,93 @@ let availableIdentities = [];
...
+  form.addEventListener('submit', (e) => {
+    e.preventDefault();
+    const form = e.target;
+
+    const formData = new FormData(form);
+    const data = Object.fromEntries(formData.entries());
+    let isValid = true;
+    if (displayName.value.trim().length === 0) {
+      displayName.classList.add('error');
+      displayNameError.textContent = 'You must provide a display name';
```
[the remark this answers is not in the document]
Comment (svdwyer) [reply src/popup/popup.js]: ✔️

File: src/popup/popup.js [hunk 4]
```diff
@@ -2,6 +2,93 @@ let availableIdentities = [];
...
+      isValid = false;
+    }
+    if (webID.value.trim().length === 0 && idp.value.trim().length === 0) {
+      webID.classList.add('error');
+      idp.classList.add('error');
+      ipdError.textContent = 'Please provide either an Identity Provider or WebID';
```
[the remark this answers is not in the document]
Comment (svdwyer) [reply src/popup/popup.js]: ✔️

File: src/popup/popup.js [hunk 5]
```diff
@@ -2,6 +2,93 @@ let availableIdentities = [];
...
+      webidError.textContent = 'Please provide either an Identity Provider or WebID';
```
[the remark this answers is not in the document]
Comment (svdwyer) [reply src/popup/popup.js]: ✔️

File: src/popup/popup.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (pheyvaer) [reply src/popup/popup.html:106]: Yeah, that's fine.

**Review by pheyvaer — APPROVED** [review]
